File: src/main/java/systems/conduit/main/console/MessageFactory.java
package systems.conduit.main.console;

import org.apache.logging.log4j.message.Message;
import org.apache.logging.log4j.message.StringFormattedMessage;

public class MessageFactory implements org.apache.logging.log4j.message.MessageFactory {

    private static ColorReplacer colorReplacer = new ColorReplacer();

    @Override
    public Message newMessage(Object message) {
        return new StringFormattedMessage(colorReplacer.apply(message.toString()));
    }

    @Override
    public Message newMessage(String message) {
        return new StringFormattedMessage(colorReplacer.apply(message));
    }

    @Override
    public Message newMessage(String message, Object... params) {
        return new StringFormattedMessage(colorReplacer.apply(message), params);
    }
}

File: src/main/kotlin/systems/conduit/core/console/MessageFactory.kt
package systems.conduit.core.console

import org.apache.logging.log4j.message.Message
import org.apache.logging.log4j.message.MessageFactory
import org.apache.logging.log4j.message.StringFormattedMessage

class MessageFactory: MessageFactory {

    override fun newMessage(message: Any): Message {
        return StringFormattedMessage(colorReplacer.apply(message.toString()))
    }

    override fun newMessage(message: String): Message {
        return StringFormattedMessage(colorReplacer.apply(message))
    }

    override fun newMessage(message: String, vararg params: Any): Message {
        return StringFormattedMessage(colorReplacer.apply(message), *params)
    }

    companion object {
        private val colorReplacer = ConsoleColorUtil()
    }
}
